Add section about building a custom Ubuntu Core image (#22)
canonical/ubuntu-for-jetson: docs/how-to/build-core-image.rst

* ubuntu-core-22-tegra-jetson-edge
* ubuntu-core-22-tegra-jetson-dangerous
* ubuntu-core-22-tegra-jetson-dangerous-candidate
* ubuntu-core-22-tegra-jetson-dangerous-beta
* ubuntu-core-22-tegra-jetson-dangerous-edge

.. note:: The model assertions that have ``-dangerous`` in their name are of ``grade: dangerous`` and can be used to make modifications to the image before installation. This should **not** be used in production!

The model assertions can be fetched and saved with the following command:

.. code-block:: bash

   snap known --remote model authority-id=canonical series=16 brand-id=canonical model=<model-name> > ubuntu-core-jetson.model

Alternatively, the model assertions can be downloaded in JSON format from the `official model repository`_. However, they need to be converted into ``.model`` files by `signing them`_ before they can be passed to the ``ubuntu-image`` command.

.. _official model repository: https://github.com/canonical/models/tree/master/devices/nvidia/jetson
.. _signing them: https://documentation.ubuntu.com/core/tutorials/build-your-first-image/sign-the-model/#ref-sign-the-model-sign-the-model

Build the image
---------------

> In order to build the image, we simply pass the model file to ``ubuntu-image``.

Reviewer: I feel a little bit strange to start using 'we' right here 
```suggestion
To build the image, simply provide the model file to ``ubuntu-image``.
```

Author: I changed this to:
```
In order to build the image, the model file simply needs to be passed to the ``ubuntu-image snap`` command.
```

Reviewer: It still looks a bit convoluted to me, but it's not a big deal

.. code-block:: bash

   ubuntu-image snap ubuntu-core-jetson.model

> The resulting image will be named ``pc.img`` and can be installed by following :doc:`documentation on how to program an ubuntu image <flash>`

Reviewer: ```suggestion
The generated image will be named ``pc.img`` and can be installed by following the :doc:`documentation for programming an Ubuntu image <flash>`.
The resulting image will be named ``pc.img`` and can be installed by following :doc:`documentation on how to program an ubuntu image <flash>`
```

Author: I think I prefer "resulting" to "generated" in this context.

I would keep the "how to" part in the documentation link as it's in the how-to section of the documentation. Maybe it could be rephrased to:
```
and can be installed by following the steps described in :doc:`How to Program an Ubuntu image`.
```
